Explorer/Assets/Scripts/Global/Dynamic/RealmNavigator.cs

```diff
@@ -90,5 +112,37 @@ public async UniTask TeleportToParcelAsync(Vector2Int parcel, CancellationToken
                 await waitForSceneReadiness.ToUniTask(ct);
             }, ct);
         }
+
+        private async UniTask GenerateWorldTerrainAsync(uint worldSeed, CancellationToken ct)
+        {
+            FixedScenePointers scenePointers = realmController.GlobalWorld.EcsWorld.Get<FixedScenePointers>(realmController.RealmEntity);
+            await UniTask.WaitUntil(() => scenePointers.AllPromisesResolved, cancellationToken: ct);
+
+            var decodedParcels = new List<int2>();
+
+            foreach (AssetPromise<SceneEntityDefinition, GetSceneDefinition> promise in scenePointers.Promises)
+            foreach (Vector2Int parcel in promise.Result.Value.Asset.metadata.scene.DecodedParcels)
+                decodedParcels.Add(parcel.ToInt2());
+
+            var ownedParcels = new NativeParallelHashSet<int2>(decodedParcels.Count, AllocatorManager.Persistent);
+
+            foreach (int2 parcel in decodedParcels)
+                ownedParcels.Add(parcel);
+
+            await landscapePlugin.WorldTerrainGenerator.GenerateTerrainAsync(ownedParcels, worldSeed, cancellationToken: ct);
+            ownedParcels.Dispose();
+        }
+
+        private void SwitchMiscVisibility(bool isVisible)
+        {
+            // isVisible
+            mapRenderer.SetSharedLayer(MapLayer.PlayerMarker, isVisible);
+            landscapePlugin.TerrainGenerator.SwitchVisibility(isVisible);
+            landscapePlugin.landscapeData.Value.showSatelliteView = isVisible;
```

Comment:
It's not a catastrophe but re-assigning the values to the SO is a bad sign. I know we already use it in different places but maybe you have a better idea to fix it quickly?

Reply:
What do you think about creating a Singleton Component for the satellite view state?
So we can toggle its visibility. 

Or we could just go the direct approach and create the Satellite view as another dependency and just have `ISatelliteView.Show()` `ISatelliteView.Hide()` `ISatelliteView().LoadAsync()`. I think this one is even better since the SatelliteView depends on the loading of the map textures.